File: Python/Questions/Number_of_Beams.py
def numberOfBeams(bank):
    n = len(bank)
    DeviceCount = 0
    result = 0
    
    for i in range(n):
        currentDevice = 0
        for ch in bank[i]:
            if ch == '1':
                currentDevice += 1
        
        result += currentDevice * DeviceCount
        
        if currentDevice != 0:
            DeviceCount = currentDevice
    
    return result

File: C++/Questions/Number_ofBeams.cpp
class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        int n = bank.size();
        int DeviceCount = 0;
        int result = 0;
        for(int i = 0; i < n; i++){
            int currentDevice = 0;
            for(char &ch : bank[i]) {
                if(ch == '1'){
                    currentDevice++;
                }
            }
            result += (currentDevice * DeviceCount);

            if(currentDevice != 0){
                DeviceCount = currentDevice;

            }
        }
        return result;
        
    }
};